### PR TITLE
feat(LINGUIST-294): Add delete message and delete conversation endpoints

### PR DESCRIPTION
**Pull request by tolgaozgun**

### CODE REVIEW

**Review by selimcanglr — DISMISSED** [review]

File: chat/views/clear_conversation.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    
+    if not conversation_id:
+        return generate_error_response(400, "Conversation ID is required")
+    
+    # Get the conversation id that matches the email if exists
+    conversation_exists = Conversation.objects.filter(id=conversation_id).exists()
+    
+    # Now get the last five messages from the conversations
+    if not conversation_exists:
+        return generate_error_response(400, "Conversation does not exist")
```
Comment (KardelenCeren) [inline chat/views/clear_conversation.py]: Can we return the error code 404 here instead of 400? (and in other files too)

**Review by KardelenCeren — CHANGES_REQUESTED** [review]
Eline sağlık ama maybe we should change the error codes here? 404 for not found, and _maybe_ 401 for authentication failure (in all files)

File: chat/views/clear_conversation.py (hunk 1 above)
Comment (can-aslan) [reply chat/views/clear_conversation.py]: Good recommendation, I agree

File: chat/views/clear_conversation.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            examples={
+                "application/json": {
+                    "timestamp": "2021-08-30 14:00:00",
+                    "status": 400,
+                    "msg": "Authentication is required"
+                }
+            }
+        ),
+        "404": openapi.Response(
+            description="Bad request",
```
Comment (KardelenCeren) [inline chat/views/clear_conversation.py]: Not found :)

File: chat/views/delete_conversation.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return generate_error_response(400, "Authentication is required")
+    
+    if not conversation_id:
+        return generate_error_response(400, "Conversation ID is required")
+    
+    conversation_exists = Conversation.objects.filter(id=conversation_id).exists()
+    
+    # Now get the last five messages from the conversations
+    if not conversation_exists:
+        return generate_error_response(400, "Conversation does not exist")
```
Comment (KardelenCeren) [inline chat/views/delete_conversation.py]: 404 would be better

File: chat/views/delete_message.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        return generate_error_response(400, "Authentication is required")
+    
+    if not message_id:
+        return generate_error_response(400, "Message ID is required")
+    
+    message_exists = Message.objects.filter(id=message_id).exists()
+    
+    # Now get the last five messages from the conversations
+    if not message_exists:
+        return generate_error_response(400, "Message does not exist")
```
Comment (KardelenCeren) [inline chat/views/delete_message.py]: 404 would be better

**Review by KardelenCeren — CHANGES_REQUESTED** [review]
Thanks! but minor stuff left

**Review by KardelenCeren — APPROVED** [review]

**Pull request merged**